klippa/src/hmtx.rs

```diff
@@ -0,0 +1,79 @@
+//! impl subset() for hmtx
+
+use crate::{Plan, Subset, SubsetError};
+use read_fonts::TopLevelTable;
+use std::cmp;
+use write_fonts::tables::{hmtx::Hmtx, hmtx::LongMetric};
+
+impl Subset for Hmtx {
+    fn subset(&mut self, plan: &mut Plan) -> Result<bool, SubsetError> {
+        let gids = &plan.glyph_ids;
+        if gids.is_empty() {
+            Err(SubsetError::SubsetTableError(Hmtx::TAG))
+        } else {
+            let num_glyphs = self.h_metrics.len() + self.left_side_bearings.len();
+            let mut num_long_metrics = cmp::min(num_glyphs, 0xFFFF_usize);
+            let last_advance = get_gid_advance(&self.h_metrics, num_glyphs - 1);
+            for gid in gids.iter().rev() {
+                let glyph_id = gid.to_u16() as usize;
+                // skip invalid gids
+                if glyph_id >= num_glyphs {
```

Comment:
I think we might need to continue in this loop, which is in reverse order, because we skip invalid gids and count the number of glyphs with identical advance for valid gids.

Reply:
Oh sorry I didn't notice this was rerversed order, in which case yeah no change is needed.